tests/xglm/test_modeling_tf_xglm.py

```diff
@@ -0,0 +1,427 @@
+
+        inputs_non_padded = tokenizer(sentences[0], return_tensors="tf").input_ids
+        output_non_padded = model.generate(input_ids=inputs_non_padded)
+
+        num_paddings = (
+            inputs_non_padded.shape[-1]
+            - tf.math.reduce_sum(tf.cast(inputs["attention_mask"][-1], dtype=tf.int64)).numpy()
+        )
+        inputs_padded = tokenizer(sentences[1], return_tensors="tf").input_ids
+        output_padded = model.generate(input_ids=inputs_padded, max_length=model.config.max_length - num_paddings)
+
+        batch_out_sentence = tokenizer.batch_decode(outputs, skip_special_tokens=True)
+        non_padded_sentence = tokenizer.decode(output_non_padded[0], skip_special_tokens=True)
+        padded_sentence = tokenizer.decode(output_padded[0], skip_special_tokens=True)
+
+        expected_output_sentence = [
+            "Hello, my dog is a little bit of a shy one, but he is very friendly",
+            "Today, I am going to share with you a few of my favorite things",
+        ]
+        self.assertListEqual(expected_output_sentence, batch_out_sentence)
+        self.assertListEqual(expected_output_sentence, [non_padded_sentence, padded_sentence])
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in TF_XGLM_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = TFXGLMModel.from_pretrained(model_name, from_pt=True)
+            self.assertIsNotNone(model)
+
+
+@require_tf
+class TFXGLMModelLanguageGenerationTest(unittest.TestCase):
+    @slow
+    def test_lm_generate_xglm(self, verify_outputs=True):
+        model = TFXGLMForCausalLM.from_pretrained("facebook/xglm-564M", from_pt=True)
+        input_ids = tf.convert_to_tensor([2, 268, 9865], dtype=tf.int32)  # The dog
+        # </s> The dog is a very friendly dog. He is very affectionate and loves to play with other
+        # fmt: off
+        expected_output_ids = [2, 268, 9865, 67, 11, 1988, 57252, 9865, 5, 984, 67, 1988, 213838, 1658, 53, 70446, 33, 6657, 278, 1581]
+        # fmt: on
+        output_ids = model.generate(input_ids, do_sample=False, num_beams=1)
+        if verify_outputs:
+            self.assertListEqual(output_ids[0].tolist(), expected_output_ids)
+
+    @slow
+    def test_xglm_sample(self):
+        tokenizer = XGLMTokenizer.from_pretrained("facebook/xglm-564M")
+        model = TFXGLMForCausalLM.from_pretrained("facebook/xglm-564M", from_pt=True)
+
+        tf.random.set_seed(0)
+        tokenized = tokenizer("Today is a nice day and", return_tensors="tf")
+        input_ids = tokenized.input_ids
+        output_ids = model.generate(input_ids, do_sample=True, num_beams=1)
+        output_str = tokenizer.decode(output_ids[0], skip_special_tokens=True)
+
+        EXPECTED_OUTPUT_STR = "Today is a nice day and the sun is shining. A nice day with warm rainy"
+        self.assertEqual(output_str, EXPECTED_OUTPUT_STR)
+
+    @slow
+    def test_xglm_sample_max_time(self):
```

Comment:
You can remove this test -- the `max_time` feature in generate is not implemented on the TF side :)